cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1145,9 +1160,21 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
             return getReconciliationStateOfConfigMap(zkCluster, zkMetricsAndLogsConfigMap, this::withZkAncillaryCmChanged);
         }
 
-        Future<ReconciliationState> zkNodesSecret() {
-            return withVoid(secretOperations.reconcile(namespace, ZookeeperCluster.nodesSecretName(name),
-                    zkCluster.generateNodesSecret(clusterCa, kafkaAssembly)));
+        Future<ReconciliationState> zkNodesSecret(Supplier<Date> dateSupplier) {
+            Promise<ReconciliationState> resultPromise = Promise.promise();
+
+            secretOperations.reconcile(namespace, ZookeeperCluster.nodesSecretName(name),
+                    zkCluster.generateNodesSecret(clusterCa, kafkaAssembly, isMaintenanceTimeWindowsSatisfied(dateSupplier)))
+                    .setHandler(res -> {
+                        if (res.succeeded()) {
+                            isZookeeperCertsChanged = res.result() instanceof ReconcileResult.Patched;
+                            resultPromise.complete(this);
+                        } else {
+                            resultPromise.fail(res.cause());
+                        }
+                    });
```

Comment:
I _think_ you can replace the `setHandler()` with:
```
map(res -> {
isZookeeperCertsChanged = res.result() instanceof ReconcileResult.Patched;
return this;
})
```
That would also avoid the need for the `resultPromise` because you should just return the overall result of the `secretOperations` expression.